src/Components/EntityState.cs

```diff
@@ -0,0 +1,8 @@
+﻿namespace Components
+{
+	public interface EntityState
+	{
+		bool TryAdd<Component>(Component component);
+		Component Get<Component>();
```

Comment:
thought: Can you think of a scenario of where you don't know the component at compile time, thus needing an overload that accepts a `Type`?